src/test/regress/expected/multi_mx_function_call_delegation.out

```diff
@@ -1,148 +1,342 @@
+DEBUG:  pushing down the function call
+DEBUG:  warning
+DETAIL:  WARNING from localhost:57638
+ERROR:  error
+CONTEXT:  while executing command on localhost:57638
+PL/pgSQL function multi_mx_select_func.mx_call_proc_raise(integer) line 4 at RAISE
+-- Test that we don't propagate to non-metadata worker nodes
+SELECT stop_metadata_sync_to_node('localhost', :worker_1_port);
+ stop_metadata_sync_to_node 
+----------------------------
+ 
+(1 row)
+
+SELECT stop_metadata_sync_to_node('localhost', :worker_2_port);
+ stop_metadata_sync_to_node 
+----------------------------
+ 
+(1 row)
+
+SELECT multi_mx_select_func.mx_dist_func(2, 0);
+DEBUG:  the worker node does not have metadata
+DEBUG:  generating subplan 21_1 for subquery SELECT sum((t1.val OPERATOR(pg_catalog.+) t2.val)) AS sum FROM (multi_mx_select_func.mx_func_dist_table_1 t1 JOIN multi_mx_select_func.mx_func_dist_table_2 t2 ON ((t1.id OPERATOR(pg_catalog.=) t2.id)))
+CONTEXT:  SQL statement "SELECT y + (SELECT sum(t1.val + t2.val) from multi_mx_select_func.mx_func_dist_table_1 t1 join multi_mx_select_func.mx_func_dist_table_2 t2 on t1.id = t2.id)"
+PL/pgSQL function mx_dist_func(integer,integer) line 8 at assignment
+DEBUG:  Plan 21 query after replacing subqueries and CTEs: SELECT (3 OPERATOR(pg_catalog.+) (SELECT intermediate_result.sum FROM read_intermediate_result('21_1'::text, 'binary'::citus_copy_format) intermediate_result(sum bigint)))
+CONTEXT:  SQL statement "SELECT y + (SELECT sum(t1.val + t2.val) from multi_mx_select_func.mx_func_dist_table_1 t1 join multi_mx_select_func.mx_func_dist_table_2 t2 on t1.id = t2.id)"
+PL/pgSQL function mx_dist_func(integer,integer) line 8 at assignment
+ mx_dist_func 
 --------------
- (2,2)
+           29
 (1 row)
 
-select mx_call_dist_object('mx_call_func', 'mx_call_dist_table'::regclass, 1);
- mx_call_dist_object 
----------------------
+SET client_min_messages TO NOTICE;
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+ start_metadata_sync_to_node 
+-----------------------------
+ 
+(1 row)
+
+SELECT start_metadata_sync_to_node('localhost', :worker_2_port);
+ start_metadata_sync_to_node 
+-----------------------------
+ 
+(1 row)
+
+SET client_min_messages TO DEBUG1;
+--
+-- Test non-const parameter values
+--
+CREATE FUNCTION mx_call_add(int, int) RETURNS int
+    AS 'SELECT $1 + $2;' LANGUAGE SQL IMMUTABLE;
+SELECT create_distributed_function('mx_call_add(int,int)', '$1');
+DEBUG:  switching to sequential query execution mode
+DETAIL:  A distributed function is created. To make sure subsequent commands see the type correctly we need to make sure to use only one connection for all future commands
+ create_distributed_function 
+-----------------------------
  
 (1 row)
 
--- test non Const distribution parameter
-select mx_call_func(2, mx_call_add(3, 4));
+-- non-const distribution parameters cannot be pushed down
+SELECT multi_mx_select_func.mx_dist_func(2, mx_call_add(3, 4));
 DEBUG:  pushing down the function call
```

Comment:
@serprex is this expected?

Reply:
No, think this might be due to changing from 0-based indexing to 1-based indexing

Reply:
In the select case the planner evaluates constant (immutable) expressions.